sound/soc/soc-topology.c

```diff
@@ -881,68 +881,30 @@ static int soc_tplg_control_dbytes_create(struct soc_tplg *tplg, struct snd_kcon
 
 static int soc_tplg_dbytes_create(struct soc_tplg *tplg, size_t size)
 {
-	struct snd_soc_tplg_bytes_control *be;
+	struct snd_kcontrol_new kc = {0};
 	struct soc_bytes_ext *sbe;
-	struct snd_kcontrol_new kc;
-	int ret = 0;
-
-	if (soc_tplg_check_elem_count(tplg,
-				      sizeof(struct snd_soc_tplg_bytes_control),
-				      1, size, "mixer bytes"))
-		return -EINVAL;
```

Comment:
is this check not relevant after moving to reusing the existing function for creating the control in the last 3 patches?

Reply:
As far as I can tell it is and was always mostly irrelevant as code only parsed one control at a time. See also commit 0db627c4f5df704d44699ed27dd81caa6d782a17 ("ASoC: topology: Remove unnecessary looping").

Only useful part is that it makes sure not to parse beyond end of the file in case someone generates topology, where payload size is corrupted, let me see how I can move it, so it stays.

Reply:
So I've just restored them in place, there is quite a few places apart from controls that miss soc_tplg_check_elem_count() check in my opinion and I feel it is best to add them in separate series.